### PR TITLE
fix #4074

### LINKED ISSUE #4074: Auth: searches don't work when signed-in via GitHub Enterprise
- Run Sourcegraph 3.4.0-rc.4
- Add a GHE external service pointing to ghe.sgdev.org
- Configure sign-in via GitHub in the management console
- Sign in via GitHub
- Run a search for `r:`
- Notice this error:

image

When I sign in using built-in auth, that search works.

I noticed the URL is http://ghe.sgdev.org/api/v3/graphql but GitHub's GraphQL API is v4.

### PR DESCRIPTION
**Pull request by beyang**

### CODE REVIEW

File: pkg/extsvc/github/client.go [hunk 1]
```diff
@@ -253,7 +253,11 @@ func (c *Client) requestGraphQL(ctx context.Context, token, query string, vars m
 		return err
 	}
 
-	req, err := http.NewRequest("POST", "/graphql", bytes.NewReader(reqBody))
+	graphqlEndpoint := "/graphql"
```
Comment (slimsag) [inline pkg/extsvc/github/client.go]: Can you add a comment above this describing why exactly this changes for github.com? e.g. describe the difference between dot com and enterprise with an example URL for each

File: pkg/extsvc/github/client.go [hunk 2]
```diff
@@ -253,7 +253,11 @@ func (c *Client) requestGraphQL(ctx context.Context, token, query string, vars m
...
+	if !c.githubDotCom {
+		graphqlEndpoint = "../graphql"
```
Comment (slimsag) [inline pkg/extsvc/github/client.go]: Does Go or GitHub handle relative paths like this? I wonder if there is a chance that in the future GitHub will regress in supporting this and it'll break auth for us.

Comment (beyang) [reply pkg/extsvc/github/client.go]: `..` is part of the URL RFC: https://tools.ietf.org/html/rfc3986

**Comment by beyang** [comment]
Merged directly

**Pull request closed without merging**